File: androidApp/src/main/java/com/mipastudio/memostamp/ui/components/StampGeometry.kt
package com.mipastudio.memostamp.ui.components

object StampGeometry {
    const val OUTPUT_WIDTH: Int = 1200
    const val OUTPUT_HEIGHT: Int = 1500

    const val ASPECT_RATIO: Float = 4f / 5f // 0.8f

    // Perforation tooth & notch ratios
    const val NOTCH_RADIUS_RATIO: Float = 0.025f
    const val NOTCH_SPACING_RATIO: Float = 0.072f

    // Standard stamp mold aspect ratio & width scale
    const val MOLD_WIDTH_RATIO: Float = 0.72f
    const val MOLD_ASPECT_RATIO: Float = 1159f / 881f // ~1.3155f

    // Cửa sổ ảnh trung tâm (Inner window ratios - 0.8 aspect ratio matching 1200x1500)
    const val INNER_LEFT_RATIO: Float = 228f / 881f
    const val INNER_TOP_RATIO: Float = 316.125f / 1159f
    const val INNER_RIGHT_RATIO: Float = 651f / 881f
    const val INNER_BOTTOM_RATIO: Float = 844.875f / 1159f
}

File: iosApp/iosApp/UI/StampGeometry.swift
import Foundation
import CoreGraphics

/// Shared Stamp Geometry specs matching Android StampGeometry 1:1.
public enum StampGeometry {
    public static let outputWidth: CGFloat = 1200
    public static let outputHeight: CGFloat = 1500

    public static let aspectRatio: CGFloat = 4.0 / 5.0 // 0.8

    // Perforation tooth & notch ratios
    public static let notchRadiusRatio: CGFloat = 0.025
    public static let notchSpacingRatio: CGFloat = 0.072

    // Standard stamp mold aspect ratio & width scale
    public static let moldWidthRatio: CGFloat = 0.72
    public static let moldAspectRatio: CGFloat = 1159.0 / 881.0 // ~1.3155

    // Inner photo window ratios
    public static let innerLeftRatio: CGFloat = 228.0 / 881.0
    public static let innerTopRatio: CGFloat = 316.125 / 1159.0
    public static let innerRightRatio: CGFloat = 651.0 / 881.0
    public static let innerBottomRatio: CGFloat = 844.875 / 1159.0
}
